src/main/java/cloud/eppo/Utils.java

```diff
@@ -16,28 +15,60 @@
 public final class Utils {
   private static final SimpleDateFormat UTC_ISO_DATE_FORMAT = buildUtcIsoDateFormat();
   private static final Logger log = LoggerFactory.getLogger(Utils.class);
+  private static final MessageDigest md = buildMd5MessageDigest();
+
+  private static MessageDigest buildMd5MessageDigest() {
+    try {
+      return MessageDigest.getInstance("MD5");
+    } catch (NoSuchAlgorithmException e) {
+      throw new RuntimeException("Error computing md5 hash", e);
+    }
+  }
```

Comment:
Instantiate the `MessageDigest` once instead of every call to `getMD5Hex()` or `getShard()`

We use `reset()` to ensure calls don't interfere with each other.